src/transformers/models/msclap/configuration_msclap.py

```diff
@@ -118,24 +77,28 @@ def __init__(
         projection_dropout_prob=0,
         initializer_factor=1.0,
         projection_dim=768,
-        projection_hidden_act='gelu', 
+        projection_hidden_act="gelu",
+        num_hidden_layers=None,
+        num_attention_heads=None,
+        vocab_size=None,
         **kwargs,
     ):
         super().__init__(**kwargs)
         if "text_encoder" not in kwargs:
-            raise ValueError("Config has to be initialized with text_encoder")
-        
-        text_encoder_config = kwargs.pop('text_encoder')
-        text_encoder_model_type = text_encoder_config.pop("model_type")
+            text_encoder_config = {"model_type": "gpt2"}
```

Comment:
Why don't we raise a ValueError in that case ? 
Seems like it would hide complexity to the user, as well as introducing undocumented behaviour!

Also, do you think we could add `text_encoder` to the named parameters and not in the `kwargs`, for more explicit usage ?